host-bin/mount-chroot

```diff
@@ -185,29 +195,30 @@ for NAME in "$@"; do
             echo 'done' 1>&2
 
             # Create key file
-            wrappedkey="`mktemp`"
-            wrappedfnek="`mktemp`"
-            addtrap "rm -f '$wrappedkey' '$wrappedfnek'"
-            echo -n "$key
+            if [ -z "$NOPASS" ]; then
+                wrappedkey="`mktemp`"
+                wrappedfnek="`mktemp`"
+                addtrap "rm -f '$wrappedkey' '$wrappedfnek'"
+                echo -n "$key
 $passphrase" | ecryptfs-wrap-passphrase "$wrappedkey" -
-            echo -n "$fnek
+                echo -n "$fnek
 $passphrase" | ecryptfs-wrap-passphrase "$wrappedfnek" -
-            unset key fnek
-            echo | cat - "$wrappedkey" "$wrappedfnek" > "$KEYFILE"
+                unset key fnek
+                echo | cat - "$wrappedkey" "$wrappedfnek" > "$KEYFILE"
+            else
+                echo > "$KEYFILE"
+                echo -n '$$$$$$$$$$$$$$$$' >> "$KEYFILE"
+                echo -n "$key" >> "$KEYFILE"
+                echo -n "$fnek" >> "$KEYFILE"
+                echo -n '$$$$$$$$$$$$$$$$' >> "$KEYFILE"
```

Comment:
The first 16 characters of a wrapped key are the hex signature, so it'd be good to consistently put the padding at the beginning of each key.

I do like this approach overall.